icu4c/source/i18n/units_router.cpp

```diff
@@ -43,15 +43,14 @@ Precision UnitsRouter::parseSkeletonToPrecision(icu::UnicodeString precisionSkel
     return result;
 }
 
-UnitsRouter::UnitsRouter(const char *inputUnitIdentifer, StringPiece region, StringPiece usage,
+UnitsRouter::UnitsRouter(StringPiece inputUnitIdentifier, StringPiece region, StringPiece usage,
                          UErrorCode &status) {
-    auto inputUnit = MeasureUnit::forIdentifier(inputUnitIdentifer, status);
-    this->init(inputUnit, region, usage, status);
+    this->init(MeasureUnit::forIdentifier(inputUnitIdentifier, status), region, usage, status);
 }
 
 UnitsRouter::UnitsRouter(MeasureUnit inputUnit, StringPiece region, StringPiece usage,
```

Comment:
Calling this already results in an unnecessary copy, unless the compiler really works magic. I would make this a const reference too.

Reply:
it should work magic :), but done!